modules/dhcp/dhcp_api.rb

```diff
@@ -8,6 +12,19 @@ class Proxy::DhcpApi < ::Sinatra::Base
 
   inject_attr :dhcp_provider, :server
 
+  def validate_arguments(params)
+    [:network, :ip_address, :from, :to, :ip].each do |ip|
+      validate_ip(params[ip]) unless params[ip].nil?
+    rescue
+      log_halt 500, "Invalid IP Address #{params[ip]}"
```

Comment:
I don't think we should treat invalid input as an internal server error. 5xx codes are server errors but in this case it's really a client error (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Status). In core we typically return HTTP 422 but in other places in the smart-proxy I've seen 400. 422 feels semantically more correct.